usdt-impl/src/record.rs

```diff
@@ -5,90 +5,125 @@
 use std::{
     collections::BTreeMap,
     ffi::CStr,
-    fs,
-    path::Path,
     ptr::{null, null_mut},
 };
 
+#[cfg(feature = "des")]
+use std::{fs, path::Path};
+
 use byteorder::{NativeEndian, ReadBytesExt};
 use dof::{Probe, Provider, Section};
-use goblin::elf;
 use libc::{c_void, Dl_info};
 
+#[cfg(feature = "des")]
+use goblin::Object;
+
 pub(crate) const PROBE_REC_VERSION: u8 = 1;
 
 /// Extract probe records from the given file, if possible.
 ///
 /// An `Err` is returned if the file not an ELF file, or if parsing the records fails in some way.
 /// `None` is returned if the file is valid, but contains no records.
+#[cfg(feature = "des")]
 pub fn extract_probe_records<P: AsRef<Path>>(file: P) -> Result<Option<Section>, crate::Error> {
     let data = fs::read(file)?;
-    // These records will only exist in ELF files
-    let object = elf::Elf::parse(&data).map_err(|_| crate::Error::InvalidFile)?;
-
-    // Try to find our special `set_dtrace_probes` section from the section headers. These may not
-    // exist, e.g., if the file has been stripped. In that case, we look for the special __start
-    // and __stop symbols themselves.
-    if let Some(section) = object
-        .section_headers
-        .iter()
-        .filter_map(|header| {
-            if let Some(result) = object.shdr_strtab.get(header.sh_name) {
-                match result {
-                    Err(_) => Some(Err(crate::Error::InvalidFile)),
-                    Ok(name) => {
-                        if name == "set_dtrace_probes" {
-                            Some(Ok(header))
-                        } else {
-                            None
+    match Object::parse(&data).map_err(|_| crate::Error::InvalidFile)? {
+        Object::Elf(object) => {
+            // Try to find our special `set_dtrace_probes` section from the section headers. These may not
+            // exist, e.g., if the file has been stripped. In that case, we look for the special __start
+            // and __stop symbols themselves.
```

Comment:
I haven't seen this specific pathology -- have you see an instance where the probe section is gone entirely? The errors I've seen produce a zero-sized section.

Reply:
No, this was unrelated to the empty-section issue. I was just being defensive against the possibility of a stripped binary.